lib/ansible/utils/encrypt.py

```diff
@@ -75,6 +108,90 @@ def __init__(self, algorithm):
         self.algorithm = algorithm
 
 
+class CryptHash(BaseHash):
+    def __init__(self, algorithm):
+        super(CryptHash, self).__init__(algorithm)
+
+        if not HAS_CRYPT:
+            raise AnsibleError("crypt cannot be used as the 'libcrypt' library is not installed or is unusable.", orig_exc=CRYPT_E)
```

Comment:
Let's also set it as a direct cause:
```suggestion
            raise AnsibleError("crypt cannot be used as the 'libcrypt' library is not installed or is unusable.", orig_exc=CRYPT_E) from CRYPT_E
```